gnomad_qc/v4/annotations/generate_freq_genomes.py

```diff
@@ -1168,16 +1203,211 @@ def get_age_distribution(
+            # caused the duplication of a single site that changed position during the
+            # minrep. This results in 2 of the Han samples being given NA instead of
+            # 0/0 as their GT. To fix this we are using the VDS to directly get the
+            # genotypes for the pop_diff samples from the full v3 VDS.
+            {"locus": hl.parse_locus("chr16:89834177"), "alleles": ["G", "GGCC"]},
+            {"locus": hl.parse_locus("chr16:89834177"), "alleles": ["GCC", "G"]},
+            {
+                "locus": hl.parse_locus("chr16:89834177"),
+                "alleles": [
+                    "GCCTGGATAAGCATAGCCCGTGTGAATCTGTGAACCTGCCTGTGCTCACGGTTGGCCGTCGTAGAAGCA",
+                    "G",
+                ],
+            },
+            {"locus": hl.parse_locus("chr22:22399516"), "alleles": ["G", "A"]},
+        ]
+    ).key_by("locus", "alleles")
+
+    # Filter to only the intervals that have a variant in the patch HT so not all the
+    # VDS needs to be loaded.
+    v3_vds = hl.vds.filter_intervals(
+        v3_vds, [hl.parse_locus_interval(f"chr{i}") for i in [1, 7, 9, 11, 12, 16, 22]]
+    )
+
+    pop_diff_patch_ht = compute_an_by_group_membership(
+        v3_vds,
+        pop_diff_group_membership_ht,
+        pop_diff_patch_ht,
+        include_full_call_stats=True,
+    ).checkpoint(new_temp_file("pop_diff_89834177", "ht"))
+
+    # Frequency groups that need a patch for variant chr9-93596925-ATTTTTTTTTTTTTT-A.
+    # In the original v3.1 a single Han sample had a missing GT at this variant. In the
+    # computation for v4.0, the _het_non_ref of True caused it to be given a 0/1 (the
+    # hl.if_else statement gets a False, so it defaults to the input GT. The original v3
+    # code didn't include the _het_non_ref check, and therefore it would have evaluated
+    # to missing.
+    freq_groups = [
+        {"group": "adj", "subset": "hgdp", "gen_anc": "han"},
+        {"group": "adj", "subset": "hgdp"},
+    ]
+    pop_diff_patch_ht = pop_diff_patch_ht.annotate(
+        freq=hl.map(
+            lambda x, m: hl.if_else(
+                (pop_diff_patch_ht.locus == hl.parse_locus("chr9:93596925"))
+                & (pop_diff_patch_ht.alleles == hl.array(["ATTTTTTTTTTTTTT", "A"]))
+                & freq_groups.contains(m),
+                hl.struct(
+                    AC=x.AC - 1,
+                    AF=(x.AC - 1) / (x.AN - 2),
+                    AN=x.AN - 2,
+                    homozygote_count=x.homozygote_count,
+                ),
+                x,
+            ),
+            pop_diff_patch_ht.freq,
+            pop_diff_patch_ht.freq_meta,
+        )
+    )
+
+    # Frequency groups that need a patch for variant chr13-20656122-T-TAAAAAAAAGAA.
+    # This variant has an AC raw of 1 and AC adj of 2 because one of the TGP samples
+    # being removed has a raw GT 0/1, but adj GT of None because it fails adj. In v3.1
+    # it was None for both and therefore shouldn't be subtracted from raw either.
```

Comment:
```suggestion
    # being removed has a raw of GT 0/1 but adj GT of None because it fails adj. In v3.1 it was None for both and therefore shouldn't be subtracted from raw either.
```